### PR TITLE
Add variable to set number of parallel deployments

### PR DESCRIPTION
**Pull request by firyx**
# PULL REQUEST

## Overview

Till now we have performed deployments sequentially (rolling updates) one server at a time.

This PR adds an optional variable `parallel_deploys` that can set number of servers to deploy to in parallel.

## Checklist
Review and complete the checklist to ensure that the PR is complete before assigned to an approver.
 - [x] All new methods or updated methods have clear docstrings
 - [x] Testing added or updated for new methods
 - [x] Verify if any changes impact the WebPortal Health Checks
 - [x] Approriate documentation updated
 - [ ] Changelog file created

### CODE REVIEW

File: playbooks/portals-deploy.yml [hunk 1]
```diff
@@ -28,30 +36,25 @@
   gather_facts: False
 
   # Limit concurrency
-  serial: 1
+  serial: "{{ parallel_deploys | default(1) }}"
```
Comment (ro-tex) [inline playbooks/portals-deploy.yml:39]: Can we rename this var to `parallel`? `serial` sounds like a boolean flag while `parallel` sounds like something that controls how many parallel streams of deployments we have. Which functions the same way for `parallel: 1`.

Comment (MSevey) [reply playbooks/portals-deploy.yml:39]: I don't think so. Fil can confirm but `serial` is an ansible builtin, not something we are defining.

Comment (ro-tex) [reply playbooks/portals-deploy.yml:39]: I also don't quite understand the interaction between the two `serial` vars. When would we want to set the first `serial` to something other than 1?

**Review by MSevey — APPROVED** [review]

File: playbooks/portals-deploy.yml [hunk 2]
```diff
@@ -2,19 +2,27 @@
...
+    max_hosts: "{{ groups['webportals'] | length - 1 }}"
+
     # Default batch config (one batch with all selected hosts will be created),
     # can be overriden with cli flag --extra-vars/-e
     batch_size: 1
     batch_number: 1
 
   tasks:
+    # Check '--limit' is used
+    - name: Fail if you are targeting all dev and prod webportals
```
Comment (ro-tex) [inline playbooks/portals-deploy.yml:23]: It only now occurs to me that if we're targeting all prod portals this won't fail. Let's change it to fail when we target all prod portals (or at least as many portals as we have in the prod group).

File: playbooks/portals-deploy.yml (hunk 1 above)
Comment (ro-tex) [reply playbooks/portals-deploy.yml:39]: > I don't think so. Fil can confirm but `serial` is an ansible builtin, not something we are defining.

Oh, that would make sense.

Comment (firyx) [reply playbooks/portals-deploy.yml:39]: `serial` is a built-in Ansible variable. Our name is `parallel_deploys`.

File: playbooks/portals-deploy.yml (hunk 2 above)
Comment (firyx) [reply playbooks/portals-deploy.yml:23]: Updated.

**Review by ro-tex — APPROVED** [review]

**Review by MSevey — APPROVED** [review]

**Pull request merged**